### PR TITLE
[GEOS-11221] mkdocs preflight rst fixes

### PR DESCRIPTION
**Pull request by jodygarnett**
[GEOS-11221](https://osgeo-org.atlassian.net/browse/GEOS-11221) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geoserver&utm_campaign=badge_info)

Proposal: [GSIP-221](https://github.com/geoserver/geoserver/wiki/GSIP-221)

The script I created to convert sphinx-build content does not handle everything. It is using pandoc to convert individual files, and brute force to handle all the sphinx-build functionality between files.

This pull requests cleans up the reStructuredText content prior to conversion:

* [x] inline images are not supported by pandoc
* [x] pandoc was unhappy with `_` use assuming wiki link, many constants are now quoted as code
* [x] pandoc wanted `|` to be substitutions, so some plaintext examples were quoted as code  
* [x] anchor headings with trailing whitespace
* [x] headings with trailing whitespace
* [x] broken relative links 

All good things, the inline images is the only content regression.

This is done as a bunch of individual commits as problems were encountered. Hopefully to make back porting easier. 

**Progress tracking spreadsheet**: https://docs.google.com/spreadsheets/d/1HMqUpTirEwwSQfeNYikPjU0aFMl0-W4Xj44cg8V_VJA/edit?usp=sharing

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geoserver.org/latest/en/developer/policies/committing.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [x] [Documentation](https://github.com/geoserver/geoserver/tree/main/doc/en/user/source) has been updated (if change is visible to end users).
- [ ] The [REST API docs](https://github.com/geoserver/geoserver/tree/main/doc/en/api/1.0.0) have been updated (when changing configuration objects or the REST controllers).
- [ ] There is an issue in the [GeoServer Jira](https://osgeo-org.atlassian.net/browse/GEOS/summary) (except for changes that do not affect administrators or end users in any way).
- [ ] Commit message(s) must be in the form ``[GEOS-XYZWV] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] Each commit has a single objective (if there are multiple commits, each has a separate JIRA ticket describing its goal).

### CODE REVIEW

**Comment by jodygarnett** [comment]
Navigation generation worked, down to the following broken links in `rst` files:
```
INFO    -  Doc file 'community/cog/mosaic.md' contains an unrecognized relative link 'src/modisvi/time', it was left as is.
INFO    -  Doc file 'extensions/importer/guireference.md' contains an unrecognized relative link '../../geoserver/data/database/', it was left as is.
INFO    -  Doc file 'production/linuxscript.md' contains an unrecognized relative link 'scripts/geoserver_deb', it was left as is.
INFO    -  Doc file 'production/linuxscript.md' contains an unrecognized relative link 'scripts/geoserver_suse', it was left as is.
INFO    -  Doc file 'production/linuxscript.md' contains an unrecognized relative link 'scripts/geoserver_tomcat', it was left as is.
INFO    -  Doc file 'security/tutorials/digest/index.md' contains an unrecognized relative link 'Digest%20access%20authentication', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_time-elevationseries.md' contains an unrecognized relative link 'src/ingestion', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_time-elevationseries.md' contains an unrecognized relative link 'src/elevation', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_timeseries.md' contains an unrecognized relative link 'src/ingestion', it was left as is.
```

**Comment by jodygarnett** [comment]
KNOWN ISSUE: uiconfiguration.md is generating with one broken link to content on the same page, happening in pandoc

```
docs/extensions/metadata/user.md
[WARNING] Reference not found for 'values' at target/convert/extensions/metadata/uiconfiguration.tmp.prep.rst_chunk_chunk line 1 column 10
docs/extensions/metadata/uiconfiguration.md
```

**Comment by jodygarnett** [comment]
KNOWN ISSUE: Some include files are providing warnings but seem to be working okay:
```
INFO    -  Doc file 'security/tutorials/digest/index.md' contains an unrecognized relative link 'Digest%20access%20authentication', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_time-elevationseries.md' contains an unrecognized relative link 'src/ingestion', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_time-elevationseries.md' contains an unrecognized relative link 'src/elevation', it was left as is.
INFO    -  Doc file 'tutorials/imagemosaic_timeseries/imagemosaic_timeseries.md' contains an unrecognized relative link 'src/ingestion', it was left as is.
```

File: doc/en/user/source/styling/workshop/design/symbology.rst [hunk 1]
```diff
@@ -77,23 +77,19 @@ Selection of an appropriate color palette is difficult, with a tension between w
...
+      Sequential
+   
+   .. figure:: img/color_diverging.png
+      
+      Diverging
+    
+   .. figure:: img/color_qualatative.png
+   
+      Qualitative
+
```
Comment (jodygarnett) [inline doc/en/user/source/styling/workshop/design/symbology.rst:92]: This is the only destructive change

**Comment by aaime** [comment]
Had a quick a random look and noticed this:

image

in the raster symbolizer page: https://jodygarnett.github.io/geoserver/styling/sld/reference/rastersymbolizer/

**Comment by jodygarnett** [comment]
Yeah now is the time for A/B testing:

Checking what you found the `rst` has:
```
    - <Opacity>
    - <ColorMap>
    - <ChannelSelection>
    - <ContrastEnhancement>
    - <ShadedRelief> *
    - <OverlapBehavior> *
    - <ImageOutline> *
```

Fix will be to escape these as literals ``<Opacity>`` ...

**Comment by AlexGacon** [comment]
On doc/en/user/source/installation/linux.rst, line 13, perhaps you should replace __, by _, to fix the display

**Comment by petersmythe** [comment]
@jodygarnett are you only doing the User Manual at the moment, and not the Developer Manual or Documentation Manual?

**Comment by petersmythe** [comment]
@jodygarnett have you tested that the Quick Edit button will work correctly when live?  Currently the differences are:

Old page : https ://github.com/geoserver/geoserver/**tree**/main/**doc**/en/user/**source**/installation/docker.**rst**  
New page: https ://github.com/geoserver/geoserver/**edit**/main/**docs**/en/user/**docs**/installation/docker.**md**

**Comment by jodygarnett** [comment]
> @jodygarnett have you tested that the Quick Edit button will work correctly when live? Currently the differences are:
> 
> Old page : https ://github.com/geoserver/geoserver/**tree**/main/**doc**/en/user/**source**/installation/docker.**rst** New page: https ://github.com/geoserver/geoserver/**edit**/main/**docs**/en/user/**docs**/installation/docker.**md**

That is correct, because when these docs are published as `md` files that will be the correct location.

(I have tested this effect before https://jodygarnett.github.io/translate/guide/ )

**Comment by jodygarnett** [comment]
@petersmythe I have managed to update the edit and view links to target this pull request:
- Commits start with ``rst:`` 
- Make small changes to make conflict resolution during backport easier.

**Comment by jodygarnett** [comment]
Okay I have processing of list-tables into definition lists figured out, for 90% of the cases it looks fine. For more than one row it looks like this:
image

File: doc/en/user/source/data/raster/imagemosaic/configuration.rst [hunk 2]
```diff
@@ -40,55 +40,55 @@ Primary configuration file
...
-   * - LevelsNum
-     - Y
-     - Represents the number of reduced resolution layers that we currently have for the granules of this mosaic.
+
+`Levels` 
+   *Mandatory*
+   
+   Represents the resolutions for the various levels of the granules of this mosaic.
```
Comment (jodygarnett) [inline doc/en/user/source/data/raster/imagemosaic/configuration.rst:47]: @aaime this change to imagemoasic configuration is the most significant change in this PR, I can try reverting it to determine if the now automatic grid-table conversion to definition list migration works.

image

Comment (jodygarnett) [reply doc/en/user/source/data/raster/imagemosaic/configuration.rst]: I was able to revert these manual changes, you can see the result available now. It is fine ...

Original rst from main branch: [configuration.rst](https://raw.githubusercontent.com/geoserver/geoserver/main/doc/en/user/source/data/raster/imagemosaic/configuration.rst)

image

This pull request mkdocs-preflight-rst-fixes branch rst:  [configuration.rst](https://raw.githubusercontent.com/jodygarnett/geoserver/mkdocs-preflight-rst-fixes/doc/en/user/source/data/raster/imagemosaic/configuration.rst)

image

The resulting markdown mkdocs-preflight-test branch conversion:  [configuration.md](https://raw.githubusercontent.com/jodygarnett/geoserver/mkdocs-preflight-test/doc/en/user/docs/data/raster/imagemosaic/configuration.md)

image

Comment (jodygarnett) [reply doc/en/user/source/data/raster/imagemosaic/configuration.rst]: The conversion script is leaving some things as table which you may prefer to edit as a definition list; I could do some rule like "if the cell is over 80 characters convert to a definition list :)

File: doc/en/user/source/installation/win_binary.rst [hunk 3]
```diff
@@ -32,7 +32,7 @@ Installation
 
 #. Download the archive and unpack to the directory where you would like the program to be located.
 
-   .. note:: A suggested location would be :file:`C:\\Program Files\\GeoServer`.
+   .. note:: A suggested location would be :file:`C:\\\\Program Files\\GeoServer`.
```
[the remark this answers is not in the document]
Comment (jodygarnett) [reply doc/en/user/source/installation/win_binary.rst]: Okay, sorry I cannot remember windows

**Comment by aaime** [comment]
The backport to `2.24.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
error: could not apply 4bbda28a37... rst: remove instructor nodes double directive
hint: After resolving the conflicts, mark them with
hint: "git add/rm <pathspec>", then run
hint: "git cherry-pick --continue".
hint: You can instead skip this commit with "git cherry-pick --skip".
hint: To abort and get back to the state before "git cherry-pick",
hint: run "git cherry-pick --abort".

```
</details>

<details>
<summary>stdout</summary>

```
[backport-7429-to-2.24.x 6668450b61] rst: small correction to sphinx-build content
 Author: Jody Garnett <jody.garnett@gmail.com>
 Date: Fri Dec 1 14:59:22 2023 +0900
 1 file changed, 2 insertions(+), 2 deletions(-)
Auto-merging doc/en/user/source/installation/docker.rst
[backport-7429-to-2.24.x cf7e25e5ef] rst: working on installation docs translation
 Author: Jody Garnett <jody.garnett@gmail.com>
 Date: Fri Feb 16 14:59:22 2024 -0800
 2 files changed, 7 insertions(+), 12 deletions(-)
[backport-7429-to-2.24.x c0dfd18437] rst: update external links and windows file references
 Author: Jody Garnett <jody.garnett@gmail.com>
 Date: Fri Feb 16 15:04:50 2024 -0800
 1 file changed, 1 insertion(+), 1 deletion(-)
Auto-merging doc/en/user/source/installation/docker.rst
[backport-7429-to-2.24.x 2b6b73e771] rst: fix fenced div use for docer installation page
 Author: Jody Garnett <jody.garnett@gmail.com>
 Date: Fri Feb 16 15:06:16 2024 -0800
 2 files changed, 4 insertions(+), 4 deletions(-)
Auto-merging doc/en/user/source/extensions/mapml/index.rst
CONFLICT (content): Merge conflict in doc/en/user/source/extensions/mapml/index.rst
Auto-merging doc/en/user/source/styling/workshop/css/point.rst

```
</details>

To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-2.24.x 2.24.x
# Navigate to the new working tree
cd .worktrees/backport-2.24.x
# Create a new branch
git switch --create backport-7429-to-2.24.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick e0996fe0557862428407abd8c70408c77f7aa2ed,4e187c05ca42a5372290c9a51c3b31ccfcc22e65,bb81eeecaef5fd740011c1376e6e4c59d696dd49,4920cf547619f2296d1c60c5b295d45cd57f3820,4bbda28a379ad6879ec0b551319f5bb82d6e50d4,566b90149302f766050a1908b599df7adb412f6f,4727e8ed04002da81a98475786e759463cd8911f,fef8fb825b4b425cc6688615bae5d9eca97cecd6,6fbf938621ce688a7c3d5d668208cdd82b04c2f3,2d9afe8ed50326e3cf2559c6a89deb8467cf1ffe,807b26b6f8f323b6fc705b459732bfd7b51110a2,7505b66e4abb0185c71f430fa69fbdb87cb78394,f591b5f389923c1419d9cb2cecd8b9eb07f27f6c,fbc4b50bf91aea7a8aadeb8ef925938aff90a716,0c8cc8879061d1b06ae41e4f541d258282addd6b,e0b620d7e12757b0dffed577b3bb9347fec0e622,f425286a3656f68b60de5b4430653c0a3c7f26e8,ca2e4f68de3005e41a55ef407eb298b2023f2cb1,901d173cb8e10dedefbd7751bd31b672268cc5f2,177fa6f298d2eca2bf3c62b574981f9dcfe8d544,3c65cd54df0ceaead8a2f8f4c77e97a79bd97275,f3054e7853db4624ed34f2a47d5b0b549e848bf0,5af1b34ee96459296d68da55d869deb63212fbc5,b4745b8cd9c0a6ae93ffb51e6671f081456c30b3,b576aa145b1a958c7fabf5348576689280c489db,8b0af96f6424fe9632bc75ac0867362f4e2ac4b5,1b02441fe2a73924a4fccdcc7a4fde5760f3b383,d753d07f005514a0da1a5148676aef557744fd28,96683269942cb4d6c5ef4d8fe00708a26320908c,0b34c688ec1e9b6f6b2de487e340d37a04357d1a
# Push it to GitHub
git push --set-upstream origin backport-7429-to-2.24.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-2.24.x
```
Then, create a pull request where the `base` branch is `2.24.x` and the `compare`/`head` branch is `backport-7429-to-2.24.x`.

**Pull request merged**